app_build_suite/build_steps/helm.py

```diff
@@ -435,17 +450,29 @@ def run(self, config: argparse.Namespace, context: Dict[str, Any]) -> None:
         chart_yaml_path = os.path.join(config.chart_dir, _chart_yaml)
         with open(chart_yaml_path, "r") as file:
             chart_yaml = yaml.safe_load(file)
+        original_annotations = chart_yaml.get(self._key_annotations, None)
         # try to guess the package file name. we need it for url generation in annotations
         chart_name = os.path.basename(config.chart_dir)
         context[context_key_chart_file_name] = f"{chart_name}-{context[context_key_git_version]}.tgz"
+        context[context_key_chart_full_path] = os.path.join(config.destination, context[context_key_chart_file_name])
+        # create metadata directory
+        context[context_key_meta_dir_path] = f"{context[context_key_chart_full_path]}-meta"
+        pathlib.Path(context[context_key_meta_dir_path]).mkdir(parents=True, exist_ok=True)
         # put in generated annotations
         chart_yaml[self._key_annotations] = {
             **chart_yaml.get(self._key_annotations, {}),
-            **self.additional_annotations(
-                config.catalog_base_url, context[context_key_chart_file_name], config.chart_dir
+            **self.build_file_annotations(
+                config.catalog_base_url,
+                context[context_key_chart_file_name],
+                config.chart_dir,
+                context[context_key_meta_dir_path],
             ),
         }
         # save Chart.yaml
+        if not context[context_key_changes_made] and original_annotations != chart_yaml[self._key_annotations]:
```

Comment:
should we extract this file backing up method to something like `utils.files.make_backup(file_name: str)`?

Reply:
IMO it should be fine to leave it like this. Next time someone needs this, it can be moved into `utils`